designer/server/src/main/scala/pl/touk/nussknacker/ui/process/processingtype/ScenarioParametersDeterminer.scala

```diff
@@ -24,15 +25,16 @@ object ScenarioParametersDeterminer {
       case Nil => throw new IllegalStateException(s"Empty list of components for processing type: $processingType")
       case nonEmptyList =>
         val intersection = nonEmptyList.foldLeft(ProcessingMode.all) {
-          case (acc, Some(componentProcessingModes)) => acc.intersect(componentProcessingModes)
-          case (acc, None)                           => acc
+          case (acc, AllowedProcessingModes.SetOf(componentProcessingModes)) =>
```

Comment:
we could have a `ToProcessingModes` extension class:

```scala
implicit class ToProcessingModes(allowedProcessingModes: AllowedProcessingModes) extends AnyVal {
  def toProcessingModes: Set[ProcessingMode] = ???
}
```
and doesn't have to pattern match on it here